pg_temp.py

```diff
@@ -112,7 +124,11 @@ def _setup(self, databases, verbosity, retry, tincr, initdb, postgres,
             self.pg_process = None
             self.pg_temp_dir = None
 
-            self.pg_temp_dir = tempfile.mkdtemp(prefix='pg_tmp_')
+            if dirname:
+                os.mkdir(dirname)
```

Comment:
What if the parent directory doesn't exist?  `os.makedirs(dirname)` might make more sense in that case.  There's also another concern -- what if the directory does exist?  This will throw an exception.  I think the usual approach is to try `makedirs` and then if it fails, catch it and `pass` the exception only if the `e.errno == errno.EEXIST`, otherwise re-raise.

Reply:
+1 for using makedirs. Also, the exist_ok parameter for makedirs has existed since python 3.2 and I think we should use that.  I don't think we need to worry about versions older than that.  Part of the reason the tests currently fail is we still have python 2.6 listed as a supported version. I think we should just update the list to 2.7, and 3.5+

Reply:
exist_ok was introduced in python 3.2 https://docs.python.org/3/library/os.html#os.makedirs
do you still want me to add it?

Reply:
Also, the same can be said for the other os.mkdirs below, I was just following convention. I can change the others as well.